build/Program.cs

```diff
@@ -0,0 +1,331 @@
+                            {
+                                return ApplyCommitInternal(commit, changes, repo);
+                            }
+                        }
+
+                        if (!string.IsNullOrWhiteSpace(change.Path))
+                        {
+                            if (MatchPath(change.Path))
+                            {
+                                return ApplyCommitInternal(commit, changes, repo);
+                            }
+                        }
+                    }
+                }
+
+                return true;
+            }
+
+            private string CalculateVersionNumber(string branch)
+            {
+                var version = this.Version.ToFullString();
+                
+                if (this.CommitCountSinceVersionChange == 1 && branch == "master") //master only
+                {
+                    if (this.Version.IsPrerelease)
+                    {
+                        //prerelease always needs the build counter just not on a branch name
+                        return $"{version}-{this.CommitCountSinceVersionChange:00000}";
+                    }
+                   
+                    // this is the full release happy path, first commit after changing the version number
+                    return version;
+                }
+
+                var rootSpecialVersion = "";
+
+                if (this.Version.IsPrerelease)
+                {
+                    // probably a much easy way for doing this but it work sell enough for a build script
+                    var parts = version.Split(new[] { '-' }, 2);
+                    version = parts[0];
+                    rootSpecialVersion = parts[1];
+                }
+
+                // if master and the version doesn't manually specify a prerelease tag force one on for CI builds
+                if (branch == "master")
+                {
+                    if (!this.Version.IsPrerelease)
+                    {
+                        branch = fallbackTag;
+                    }else
+                    {
+                        branch = "";
+                    }
+                }
+
+                if (rootSpecialVersion.Length > 0)
+                {
+                    rootSpecialVersion = "-" + rootSpecialVersion;
+                }
+                if (branch.Length > 0)
+                {
+                    branch = "-" + branch;
+                }
+
+                var maxLength = 20; // dotnet will fail to populate the package if the tag is > 20
+                maxLength -= rootSpecialVersion.Length; // this is a required tag
+                maxLength -= 7; // for the counter and dashes
+                
+                if(branch.Length > maxLength)
+                {
+                    branch = branch.Substring(0, maxLength);
+                }
+                
+                return $"{version}{rootSpecialVersion}{branch}-{this.CommitCountSinceVersionChange:00000}";
```

Comment:
This means that branches and PRs wont ever produce version numbers that match master ones, if you allow branches to be able to produce packages that could also be produced by master you might end up with 2 different builds producing the same version number and causing accidental confusion if anyone finds/uses the wrong file.